cmd/controller-manager/app/controller_manager.go

```diff
@@ -111,21 +112,30 @@ func Run(controllerManagerOptions *options.ControllerManagerServer) error {
 		rest.AddUserAgent(k8sKubeconfig, controllerManagerAgentName),
 	)
 	if err != nil {
-		glog.Fatalf("Invalid k8s API configuration: %v", err)
+		return fmt.Errorf("invalid Kubernetes API configuration: %v", err)
 	}
 
 	glog.V(4).Infof("Building service-catalog kubeconfig for url: %v\n", controllerManagerOptions.ServiceCatalogAPIServerURL)
+
+	var serviceCatalogKubeconfig *rest.Config
 	// Build the service-catalog kubeconfig / clientBuilder
-	serviceCatalogKubeconfig, err := clientcmd.BuildConfigFromFlags(
-		controllerManagerOptions.ServiceCatalogAPIServerURL,
-		// controllerManagerOptions.ServiceCatalogKubeconfigPath,
-		"", // TODO: tolerate missing kubeconfig
-	)
+	if controllerManagerOptions.ServiceCatalogAPIServerURL == "" && controllerManagerOptions.ServiceCatalogKubeconfigPath == "" {
+		// explicitly fall back to InClusterConfig, assuming we're talking to an API server which does aggregation
```

Comment:
Oh okay. Not obvious to me. So the `this` that `BuildConfigFromFlags` `does`, is the fallback?